src/Components/PhotoUpload/Utils/uploadFileToS3.tsx

```diff
@@ -23,16 +30,27 @@ export const uploadFileToS3 = (
     const bucket = asset.policyDocument.conditions.bucket
     const uploadURL = `https://${bucket}.s3.amazonaws.com`
 
+    const isExternalPhoto = photo.externalUrl
+
+    let file
+
+    try {
+      file = isExternalPhoto ? await externalPhotoToFile(photo) : photo.file
+    } catch (error) {
+      reject(new Error("Initializing artwork photo failed."))
+      return
+    }
```

Comment:
**Q:** I am wondering if you considered moving this logic to where the upload file method is being called to always send files here in order to avoid initializing the file here